src/streams/tls.h

```diff
@@ -20,4 +20,412 @@
  */
 extern int git_tls_stream_new(git_stream **out, const char *host, const char *port);
 
+/**
+ * A TLS cipher
+ *
+ * Only used to provide mappings from GIT_OPT_SET_SSL_CIPHERS names to
+ * what the TLS backend uses.
+ */
+typedef struct {
+	uint32_t value;
+	const char *openssl_name;
+	const char *nist_name;
+} git_tls_cipher;
+
+/**
+ * Parse a list of cipher names from cipher_list.
+ *
+ * The intended usage is to pass a pointer to a list of cipher; either the
+ * default list of ciphers (GIT_TLS_DEFAULT_CIPHERS) or the list configured by
+ * GIT_OPT_SET_SSL_CIPHERS.
+ *
+ * As this function must work with read-only data, the name is never NULL-terminated.
+ *
+ * @param out_name the cipher name
+ * @param out_len the cipher name length
+ * @param cipher_list the current position in the cipher list
+ * @return 0 on success, GIT_ITEROVER if the list is empty
+ */
+GIT_EXTERN(int) git_tls_enabled_ciphers(const char **out_name, size_t *out_len, const char **cipher_list);
```

Comment:
This interface is a bit weird. It does not have anything to do with enabled ciphers, but only acts as some kind of special-cased strtok, right?

Reply:
True. I'll rename it to `git_tls_cipher_foreach` or something.